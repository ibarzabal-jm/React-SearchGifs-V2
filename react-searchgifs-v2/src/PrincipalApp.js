import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import GifGrid from './components/GifGrid';

export const PrincipalApp = () => {

   
    
    const [categories, setCategories] = useState(['Hunter x Hunter']);

    return (
        <>
            <h1> SearchGifs V2 </h1>
            <br />
            <AddCategory setCategories = {setCategories} />
            <hr />


            <ol>
                {
                    categories.map( category => 

                        <GifGrid 
                            category={category}
                            key={category}
                            
                        />

                    )
                }
            </ol>
        </>
    )
}
