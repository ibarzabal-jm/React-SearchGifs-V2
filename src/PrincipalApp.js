import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import {GifGrid} from './components/GifGrid';

export const PrincipalApp = ( { defaultCategories = [] }) => {

   
    
    // const [categories, setCategories] = useState(['Amor']);
    const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <h1> SearchGifs V2 </h1>
            <br />
            <AddCategory setCategories = {setCategories} />
            <hr />


            <ol>
                {
                    categories.map( category => 

                        <GifGrid 
                            category={category}
                            key={category}
                            
                        />

                    )
                }
            </ol>
        </>
    )
}
