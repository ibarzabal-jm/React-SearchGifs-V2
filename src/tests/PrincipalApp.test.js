import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import {PrincipalApp} from '../PrincipalApp'

describe('Pruebas en <PrincipalApp />', () => {
    
    test('debe de mostrar el componente Correctamente ', () => {
        
        const wrapper = shallow( <PrincipalApp /> );

        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de mostrar una lista de categorÃ­as', () => {
        
        const categories = ['Hunter x Hunter', 'Dragon ball'];
        const wrapper = shallow( <PrincipalApp defaultCategories={categories} /> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
    
})
