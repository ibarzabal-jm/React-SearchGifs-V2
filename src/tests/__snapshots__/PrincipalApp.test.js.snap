// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrincipalApp /> debe de mostrar el componente Correctamente  1`] = `
<Fragment>
  <h1>
     SearchGifs V2 
  </h1>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <PrincipalApp /> debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h1>
     SearchGifs V2 
  </h1>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hunter x Hunter"
      key="Hunter x Hunter"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
